src/sInv.sol

```diff
@@ -238,16 +252,15 @@ contract sINV is ERC4626 {
      * @param exactDbrOut The exact amount of DBR to receive.
      * @param to The address that will receive the DBR.
      */
-    function buyDBR(uint exactInvIn, uint exactDbrOut, address to) external {
-        require(to != address(0), "Zero address");
-        uint DBRBalance = getDbrReserve();
+    function buyDBR(uint256 exactInvIn, uint256 exactDbrOut, address to) external nonReentrant {
+        uint256 DBRBalance = getDbrReserve();
         if(exactDbrOut > DBR.balanceOf(address(this))){
             invEscrow.claimDBR();
         }
-        uint k = getK();
-        uint DBRReserve = DBRBalance - exactDbrOut;
-        uint invReserve = k / DBRBalance + exactInvIn;
-        require(invReserve * DBRReserve >= k, "Invariant");
+        uint256 k = getK();
+        uint256 DBRReserve = DBRBalance - exactDbrOut;
+        uint256 invReserve = k / DBRBalance + exactInvIn;
+        if(invReserve * DBRReserve < k) revert Invariant();
```

Comment:
Yeah it's primarily to be flexible with rounding and because we expect most if not all interactions to come from MEV bots, though it may be preferable to add the purchase functions from the `sInvHelper` contract to the main contract to make them more gas efficient.

I'm not sure if there's a scenario where an MEV bot may specifically want to pay a few tokens more, to save the gas cost associated with calculating the exact exchange rate at blocktime, though that seems unlikely outside of a very high gas scenario.